x-pack/legacy/plugins/maps/public/connected_components/map/mb/view.js

```diff
@@ -454,6 +454,15 @@ export class MBMapContainer extends React.Component {
     addSpritesheetToMap(json, sprites, this._mbMap);
   }
 
+  _reevaluateTooltipPosition = () => {
+    // Force mapbox to ensure tooltip does not clip map boundary and move anchor when clipping occurs
+    requestAnimationFrame(() => {
+      if (this._isMounted) {
```

Comment:
I'd also add a conditional to check if `tooltipState.location` is valid. It's possible it's already been erased because of the debounce.